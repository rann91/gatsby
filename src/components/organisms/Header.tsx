/* @jsx jsx */
import { Box, Container, jsx } from 'theme-ui'
import { rem } from '../../gatsby-plugin-theme-ui'
import Branding from '../molecules/Branding'
import Navigation from '../molecules/Navigation'

const Header = () => (
  <header
    sx={{
      position: 'sticky',
      zIndex: 1,
      top: 0,
      height: [rem(65), rem(85), null, rem(105)],
      backgroundColor: 'white',
      boxShadow: '0 0 4px rgba(0, 0, 0, 0.1)'
    }}>
    <Container
      sx={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
        height: '100%'
      }}>
      <Box px={5}>
        <Branding />
      </Box>
      <Box px={5}>
        <Navigation />
      </Box>
    </Container>
  </header>
)

export default Header
