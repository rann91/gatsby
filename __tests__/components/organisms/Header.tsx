import { render } from '@testing-library/react'
import { useStaticQuery } from 'gatsby'
import React from 'react'
import Header from '../../../src/components/organisms/Header'
import navigation from '../../fixtures/navigation.json'
import siteMetadata from '../../fixtures/site-metadata.json'

beforeEach(() => {
  ;(useStaticQuery as jest.Mock)
    // Mock for use in Branding
    .mockImplementationOnce(() => siteMetadata)
    // Mock for use in Navigation
    .mockImplementationOnce(() => navigation)
})

describe('Header', () => {
  it('renders correctly', () => {
    const { container } = render(<Header />)

    expect(container).toMatchSnapshot()
  })
})
