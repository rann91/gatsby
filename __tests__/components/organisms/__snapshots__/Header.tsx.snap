// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <header
    class="css-14nc0uu-Header"
    data-testid="header"
  >
    <div
      class="css-zk5be6-Header"
    >
      <div
        class="css-i1z8pl"
      >
        <a
          class="css-13dgcig-Branding"
          data-testid="branding"
          href="/"
        >
          <div
            class="css-18ov0hz-Branding"
          >
            <div
              class="css-fhum7v-Branding"
            >
              <img
                alt="Branding Logo"
                class="css-9taffg"
                src="test-file-stub"
              />
            </div>
            <div
              class="css-vurnku"
            >
              Gatsby
            </div>
          </div>
        </a>
      </div>
      <div
        class="css-i1z8pl"
      >
        <div
          class="css-1l0o9ue-PureNavigation"
        >
          <button
            aria-label="Open Menu"
            class="css-wwcqj1"
            title="Menu"
          >
            <svg
              class="css-ipc245"
              fill="currentcolor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </button>
        </div>
        <nav
          class="css-4swsle-PureNavigation"
        >
          <ul
            class="css-13786ft-PureNavigation"
          >
            <li
              class="css-1w5hvub"
            >
              <a
                class="css-1ys58o5"
                data-testid="gatsby-link"
                href="/"
                target="_self"
              >
                Home
              </a>
            </li>
            <li
              class="css-1w5hvub"
            >
              <a
                class="css-1ys58o5"
                data-testid="gatsby-link"
                href="/about"
                target="_self"
              >
                About
              </a>
            </li>
            <li
              class="css-1w5hvub"
            >
              <a
                class="css-1ys58o5"
                data-testid="gatsby-link"
                href="/work"
                target="_self"
              >
                Work
              </a>
            </li>
            <li
              class="css-1w5hvub"
            >
              <a
                class="css-1ys58o5"
                data-testid="gatsby-link"
                href="/team"
                target="_self"
              >
                Team
              </a>
            </li>
            <li
              class="css-1w5hvub"
            >
              <a
                class="css-1ys58o5"
                data-testid="gatsby-link"
                href="/blog"
                target="_self"
              >
                Blog
              </a>
            </li>
            <li
              class="css-1w5hvub"
            >
              <a
                class="css-1ys58o5"
                data-testid="gatsby-link"
                href="/contact"
                target="_self"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;
